### PR TITLE
Test on python 3.12

### PR DESCRIPTION
**Pull request by maxnoe**

### CODE REVIEW

**Comment by maxnoe** [comment]
wohoo :fireworks:

**Review by kosack — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -43,19 +43,23 @@ jobs:
 
           - os: ubuntu-latest
             python-version: "3.10"
```
Comment (Tobychev) [inline .github/workflows/ci.yml:45]: Is there a reason why phyton 3.10 only uses pip while 3.9, 3.11, and 3.12 also tries mamba?

Comment (maxnoe) [reply .github/workflows/ci.yml:45]: No particular reason, we at some point in the past decided not to test on the full product matrix of `[python versions] x [pip, mamba]` but to only test some combinations that cover all python versions and both installation methods.

Comment (maxnoe) [reply .github/workflows/ci.yml:45]: To limit the number of parallel builds that run in the CI

**Review by Tobychev — APPROVED** [review]

**Review by kosack — APPROVED** [review]

**Pull request merged**